cmd/check.go

```diff
@@ -0,0 +1,246 @@
+
+					if major < 1 || (major == 1 && minor < 12) {
+						return ver, &versionError{"Requires 2.12 or greater."}
+					}
+
+					return ver, err
+				},
+			},
+			{
+				name:    "Terraform\t",
+				command: "terraform",
+				args:    []string{"version"},
+				docsURL: "https://www.terraform.io/downloads.html",
+				checker: func(output []byte) (string, error) {
+					ver := ""
+					re := regexp.MustCompile(`Terraform v([0-9]+)\.([0-9]+)\.([0-9]+)`)
+					m := re.FindStringSubmatch(string(output))
+					major, err := strconv.ParseInt(m[1], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+					minor, err := strconv.ParseInt(m[2], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+					patch, err := strconv.ParseInt(m[3], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+
+					ver = fmt.Sprintf("%d.%d.%d", major, minor, patch)
+
+					if major < 0 || (major == 0 && minor < 12) {
+						return ver, &versionError{"Zero requires terraform 0.12 or greater."}
+					}
+
+					return ver, err
+				},
+			},
+			{
+				name:    "jq\t\t",
+				command: "jq",
+				args:    []string{"--version"},
+				docsURL: "https://stedolan.github.io/jq/download/",
+				checker: func(output []byte) (string, error) {
+					ver := ""
+					re := regexp.MustCompile(`jq-([0-9]+)\.([0-9]+)-`)
+					m := re.FindStringSubmatch(string(output))
+					major, err := strconv.ParseInt(m[1], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+					minor, err := strconv.ParseInt(m[2], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+
+					ver = fmt.Sprintf("%d.%d", major, minor)
+
+					if major < 1 || (major == 1 && minor < 5) {
+						return ver, &versionError{"Requires jq version 1.15 or greater."}
+					}
+
+					return ver, err
+				}},
+			{
+				name:    "Git\t\t",
+				command: "git",
+				args:    []string{"version"},
+				docsURL: "https://git-scm.com/book/en/v2/Getting-Started-Installing-Git",
+				checker: func(output []byte) (string, error) {
+					ver := ""
+					re := regexp.MustCompile(`git version ([0-9]+)\.([0-9]+)\.([0-9]+)`)
+					m := re.FindStringSubmatch(string(output))
+					major, err := strconv.ParseInt(m[1], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+					minor, err := strconv.ParseInt(m[2], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+					patch, err := strconv.ParseInt(m[3], 0, 64)
+					if err != nil {
+						return ver, err
+					}
+
+					ver = fmt.Sprintf("%d.%d.%d", major, minor, patch)
+
+					if major < 2 || (major == 2 && minor < 12) {
+						return ver, &versionError{"Zero requires git version 2.12 or greater."}
+					}
+
+					return ver, err
+				}},
+		}
+
+		// Store and errors from the commands we run.
+		errors := []commandError{}
+
+		fmt.Println("Checking Zero Requirements...")
+		for _, r := range required {
+			fmt.Printf("%s", r.name)
+			out, err := exec.Command(r.command, r.args...).CombinedOutput()
+			if err != nil {
```

Comment:
Can we make a differentiation here between "the command doesn't exist" and "executing this command returned a non-zero exit code"?

Reply:
Yeah, I'll investigate this some more. I was able to infer based on the output of the command if it's missing or not as the error is consistent. We would need to check cross-platform to make sure it returns a similar error. There may also be another way to run commands that is more verbose.

E.g.
```zsh
 Command         Error                                            Info                                                     
 ---------       ---------                                        ---------                                                
kubectl version  exit status 1                                    https://kubernetes.io/docs/tasks/tools/install-kubectl/  
jq2 --version    exec: "jq2": executable file not found in $PATH  https://stedolan.github.io/jq/download/                  
git              Zero requires git version 2.12 or greater.       https://git-scm.com/book/en/v2/Getting-Started-Installing-Git% 
```